examples/systemcalls/systemcalls.c

```diff
@@ -91,9 +162,62 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   redirect standard out to a file specified by outputfile.
  *   The rest of the behaviour is same as do_exec()
  *
-*/
+*/  
+    // Integer type variable for the child pid
+    int child_pid;
+
+    // Open the output file with write permission and create if not already created
+    int file_descriptor = open(outputfile, O_WRONLY | O_CREAT | O_TRUNC, 0644); 
+
+    // Error check 1: Check if the file could be opened/created 
+    if (file_descriptor == -1) {
+        printf("Error opening/creating the file\n");
+        perror("open");
+        return false;
+    }
+
+    // Create new child process
+    switch(child_pid = fork()) {
+        // If there is an error
+        case -1: 
+            perror("Child PID:fork() command"); 
+            return false;
+
+        // If it is a child Process
+        case 0: 
+                if(dup2(file_descriptor, 1) < 0) { 
+                perror("Child PID:dup2() command"); 
+                exit(1); 
+            }
+            close(file_descriptor);
+            execv(command[0], command); 
+            perror("Child PID: execv() command"); 
+            exit(1); 
+
+        default:
+            close(file_descriptor);
+            int ret_status = 0;
+            pid_t wait_pid = waitpid(child_pid, &ret_status, 0);
+
+            if (wait_pid == -1) {
+                perror("Wait PID:wait() command");
+                return false;
+            } else {
+                // Check if process exited normally
+                if (WIFEXITED (ret_status)) {
+                    // Checking status of process
+                    if (WEXITSTATUS(ret_status) != 0) {
+                        return false;
```

Comment:
Missing cleanup in these code paths.
Also see discussion in class about avoiding duplication between these two very similar functions.